rust/serdejsonbench/src/serde_json_simple_deserializer.rs

```diff
@@ -1,17 +1,32 @@
-use std::{error::Error, fs::File, io::BufReader, path::Path};
-
-use crate::Json;
-
-pub fn read_from_file<P: AsRef<Path>>(path: P) -> Result<Vec<Json>, Box<dyn Error>> {
-    let file = File::open(path)?;
-    let reader = BufReader::new(file);
-
-    Ok(serde_json::from_reader(reader)?)
-}
-
-pub fn readone_from_file<P: AsRef<Path>>(path: P) -> Result<Json, Box<dyn Error>> {
-    let file = File::open(path)?;
-    let reader = BufReader::new(file);
-
-    Ok(serde_json::from_reader(reader)?)
-}
+use std::{
+    error::Error,
+    fs::File,
+    io::{BufReader, Read},
+    path::Path,
+};
+
+use crate::Json;
+
+pub fn read_from_file2<P: AsRef<Path>>(path: P) -> Result<Vec<Json>, Box<dyn Error>> {
+    let mut file = File::open(path)?;
+    //https://github.com/serde-rs/json/issues/160
+    let size = file.metadata().unwrap().len();
+    let mut bytes = Vec::with_capacity(size.try_into().unwrap());
+    file.read_to_end(&mut bytes).unwrap();
```

Comment:
I think you will likely still want a bufreader here too

Reply:
Usually yes but IIUC that issue https://github.com/serde-rs/json/issues/160#issuecomment-253446892 the trick is to use

```rust
File::open(path).unwrap().read_to_end(&mut bytes).unwrap();
```

Reply:
performance are back (x2 faster in rust)! 
rust: https://github.com/cboudereau/serde-bench/actions/runs/6200693064/job/16835886757#step:4:129
java: https://github.com/cboudereau/serde-bench/actions/runs/6200693064/job/16835886757#step:10:68